src/emc/plots/scatter.py

```diff
@@ -0,0 +1,103 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.colors import Normalize
+from emcpy.stats import get_linear_regression
+
+__all__ = ['scatter']
+
+
+def _density_scatter(x, y, ax=None, fig=None, sort=True, bins=20, **kwargs):
+    """
+    Creates scatter plot colored by 2d histogram
+    """
+    data, x_e, y_e = np.histogram2d(x, y, bins=bins, density=True)
+    z = interpn((0.5*(x_e[1:] + x_e[:-1]), 0.5*(y_e[1:]+y_e[:-1])),
+                data, np.vstack([x, y]).T, method="splinef2d",
+                bounds_error=False)
+    # To be sure to plot all data
+    z[np.where(np.isnan(z))] = 0.0
+    # Sort the points by density, so that the densest points are plotted last
+    if sort:
+        idx = z.argsort()
+        x, y, z = x[idx], y[idx], z[idx]
+    ax.scatter(x, y, c=z, **kwargs)
+    norm = Normalize(vmin=np.min(z), vmax=np.max(z))
+
+    return ax
+
+
+def _gen_scatter(x, y, plotopts):
+    """
+    Generate scatter plot from x and y.
+    """
+    fig = plt.figure(figsize=(8, 6))
+    ax = fig.add_subplot(111)
+
+    if plotopts['density']:
+        ax = _density_scatter(x, y, ax=ax, fig=fig, bins=[100, 100],
```

Comment:
`fig` is passed, but not used in `_density_scatter`

Reply:
You are right. Was copied from a previous build of this function that would make a figure if no fig was passed. However, this design goes all through `_gen_scatter()` and makes a fig beforehand so there is no need to pass fig to `_density_scatter()`. Fixing now.